### DISCUSSION TITLE
TypeError: context.agent.dataStoreSaveVerifiableCredential is not a function

### DISCUSSION THREAD

**Discussion by RobertoSnap** [decentralized-identity/veramo#572] [Q&A] [answered]
Trying to save a VC

`
   const vc = await this.agent.createVerifiableCredential({
      proofFormat: 'jwt',
      save: true,
      credential: {
        credentialSubject: {
          ...data,
          id: subjectDidId,
        },
        issuer: {
          id: this.issuer,
        },
      },
    });
`

Getting TypeError: TypeError: context.agent.dataStoreSaveVerifiableCredential is not a function

Have I forgot to import something?

**Comment by mirceanis** [comment, marked as answer]
you are probably missing some data store plugins in your agent setup:
```ts
agent = createAgent<
  //...
  & IDataStore
  & IDataStoreORM
>({
    // ...
    plugins: [
      // ...
      new DataStore(dbConnection),
      new DataStoreORM(dbConnection),
    ]})
```

**Reply by RobertoSnap** [reply]
Worked 👍

```ts 
import { JwtMessageHandler } from '@veramo/did-jwt';
import { W3cMessageHandler } from '@veramo/credential-w3c';
import { MessageHandler } from '@veramo/message-handler';
...
        new MessageHandler({ messageHandlers: [new JwtMessageHandler(), new W3cMessageHandler()] }),
          new DataStore(this.dbConnection),
          new DataStoreORM(this.dbConnection),
```
